python/lance/pytorch/data.py

```diff
@@ -52,26 +55,47 @@ class LanceDataset(IterableDataset):
 
     def __init__(
         self,
-        uri: Union[str, Path],
+        root: Union[str, Path],
         columns: Optional[List[str]] = None,
         filter: Optional[pc.Expression] = None,
         batch_size: Optional[int] = None,
     ):
-        self.uri = uri
+        self.root = root
         self.columns = columns if columns else []
         self.filter = filter
         self.batch_size = batch_size
+        self._dataset: pa.dataset.FileSystemDataset = None
+        self._fs: Optional[pyarrow.fs.FileSystem] = None
+        self._files: Optional[List[str]] = None
 
     def __repr__(self):
-        return f"LanceDataset(uri={self.uri})"
+        return f"LanceDataset(root={self.root})"
+
+    def _setup_dataset(self):
+        """Lazy loading dataset in different process."""
+        if self._files is not None:
+            return self._files
+
+        self._fs, _ = pyarrow.fs.FileSystem.from_uri(self.root)
+        self._files = dataset(self.root).files
+        worker_info = torch.utils.data.get_worker_info()
+        if worker_info:
+            # Split the work using at the files level for now.
```

Comment:
so just to check my understanding, this is what's forcing us to split into many smaller files right now right? Theoretically if we just have some num_rows based parallelism we could have good analytics performance and good training scan performance?

Reply:
Yes.